### PR TITLE
Rename Cookbook to How To, add Introduction, update Quick Start

### PR DESCRIPTION
**Pull request by jtoar**
This PR reorganizes the sidebar so that the doc categories reflect a kind of learning path. In the process, a new doc, Introduction is added (still a bit of a work in progress), the Quick Start is updated, and Cookbook is renamed to How To (corresponding redwoodjs.com update https://github.com/redwoodjs/redwoodjs.com/pull/1017).

Once the intro and quick start are closer to being finished, I'll update the crwa README.md as well.

### CODE REVIEW

File: docs/docs/introduction.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Introduction
```
Comment (thedavidprice) [inline docs/docs/introduction.md:1]: Suggestions:
- add a link to README.md from here, something like "For full inspiration and Vision"
- in future, we can add Roadmap
- would be great to link to the Core Team and All Contributors section in the README as well

File: docs/docs/introduction.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+Redwood is the full-stack web framework designed to help you grow from side project to startup.
+Redwood features an end-to-end development workflow that weaves together the best parts of [React](https://reactjs.org/), [GraphQL](https://graphql.org/), [Prisma](https://www.prisma.io/), [TypeScript](https://www.typescriptlang.org/), [Jest](https://jestjs.io/), and [Storybook](https://storybook.js.org/).
+
+## Getting the Most out of Redwood
+
+To get the most out of Redwood, do two things:
+
+- Start the [Tutorial](tutorial/foreword.md)
+- Join the [community](https://community.redwoodjs.com/)
```
Comment (thedavidprice) [inline docs/docs/introduction.md:11]:
```suggestion
- Join the [community](https://redwoodjs.com/community)
```

FYI: I'm going to create a page/post tomorrow that will be our Go-to for community onboarding. I'll make sure the redirect goes to the page I create (if not that post)

File: docs/docs/quick-start.md [hunk 3]
```diff
@@ -1,81 +1,74 @@
----
-sidebar_position: 2
----
-
 # Quick Start
```
Comment (thedavidprice) [inline docs/docs/quick-start.md:1]: This is a great foundation overall. I think you should add high-level info about the following:
- The Redwood CLI: help and link to doc
- Tests: CLI, files, and Doc
- Storybook: CLI, files, and Doc
- Deploy: both serverless and Fastify server, CLI setup and Doc
- Styling: Tailwind and Chakra, CLI setup

Those don't need a deep-dive, but mentioning them also communicates end-to-end, dev-to-deploy integrations

File: packages/create-redwood-app/template/README.md [hunk 4]
```diff
@@ -1,21 +1,62 @@
-# Redwood
-## Getting Started
-- [Tutorial](https://redwoodjs.com/docs/tutorial): getting started and complete overview guide.
-- [Docs](https://redwoodjs.com/docs/introduction): using the Redwood Router, handling assets and files, list of command-line tools, and more.
-- [Redwood Community](https://community.redwoodjs.com): get help, share tips and tricks, and collaborate on everything about RedwoodJS.
+# README
```
Comment (thedavidprice) [inline packages/create-redwood-app/template/README.md:1]: See my comments in the Quick-start.

Additionally, I think this should include a Section with something like "Quick Link": website, docs, twitter & newsletter, discord, discourse, etc.:
- how to stay updated: Forum Announcements and Twitter
- how to join the community
- how to submit bugs: GitHub Issues
- Contributors are welcome! Link to docs/contributing

Comment (thedavidprice) [reply packages/create-redwood-app/template/README.md:1]: FYI: sometime in April I'm going to create a Email list people can sign up for to go through the tutorial in 8 weeks. Will make for a good next step for these types of things.

**Comment by thedavidprice** [comment]
Other items to update:
- click on Docs should go to `docs/introduction`
- click on Logo should go to `https://redwoodjs.com/`


And some thoughts about improvements to the mobile view (not everything is working):

https://user-images.githubusercontent.com/2951/161406873-2d91e2d5-fb7b-481f-91cd-b718565fd36c.mp4

**Comment by jtoar** [comment]
@thedavidprice Just watched the video; agree with everything! Wanted to point out one thing about the Deployment breadcrumb link: the reason that doesn't work is there's no deployment index page: https://redwoodjs.com/docs/deploy/index. The reference one works because there is a reference index page: https://redwoodjs.com/docs/index. We could maybe make the breadcrumb link just go to the introduction.

**Comment by thedavidprice** [comment]
Hellofa job well done here @jtoar Looking really really solid all around.

**Information Architecture**
I like that the `docs/` link from the home page goes to `docs/introduction`. But if you manually enter `rwjs.com/doc` it goes to `docs/index` Now that I think about that, its probably the right UX.

I suggest EVERY top-level category — e.g. Reference, Deployment, etc. — needs an index page. Several times now I've tried to move between the sections for Deploy providers on smaller screens and the UX is effectively broken. I _expected_ the `deploy/introduction` to have links to the sections in the Right Sidebar, but it doesn't. And although I'm so used to this Doc, I even got lured into cliking on the links to the Provider websites that I intuitively thought were for sections... (I'm going to change that right. now. And will do the same for Auth.)

**Including in Final 1.0.0**
To include the new CRWA README in the 1.0.0 relase, I'm adding this next-release-patch milestone.

**Pull request merged**